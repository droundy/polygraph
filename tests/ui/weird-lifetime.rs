polygraph::schema!{
    type Tree;
    pub struct Surname(String);
    pub struct Person {
        surname: Key<Surname<'a>>,
        name: String,
    }
}

fn main() {
}
