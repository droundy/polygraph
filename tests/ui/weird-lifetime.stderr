error: Key type should be a simple table name
 --> $DIR/weird-lifetime.rs:5:29
  |
5 |         surname: Key<Surname<'a>>,
  |                             ^^^^^
